kernel/src/engine/default/mod.rs

```diff
@@ -108,6 +108,10 @@ impl<E: TaskExecutor> DefaultEngine<E> {
     pub fn get_object_store_for_url(&self, _url: &Url) -> Option<Arc<DynObjectStore>> {
         Some(self.store.clone())
     }
+
+    pub fn get_parquet_handler(&self) -> Arc<DefaultParquetHandler<E>> {
+        self.parquet.clone()
+    }
```

Comment:
I don't particularly like this but we need a way to get back the _actual_ `DefaultParquetHandler` instead of the type-erased `Arc<dyn ParquetHandler>`. I think we can do it a couple ways: either like this, or we implement as_any to allow us to just cast the types. thoughts?

Reply:
Is this because we want to be able to call `write_parquet_files`? Surely that should be part of the `ParquetHandler` api spec?

Reply:
yup! we originally didn't want to impose that requirement on engines since they _should_ elect to just write parquet files however they want. if we added it to ParquetHandler then we would be requiring an API that we (kernel) don't actually use (only the default engine). That being said, we will likely need a `write_parquet_files` API for writing checkpoints so we could just introduce the requirement here and use it in the default engine but only _really_ require it in the future once we are writing checkpoints

Reply:
mmm okay. yeah the other options are a bit... messy, downcasting is not trivial.

can we make it `pub(crate)` or does that not work?

Reply:
unfortunately doesn't work - could use developer-visibility but that's the best i could come up with for hiding it